pyiron_ontology/parser.py

```diff
@@ -165,3 +174,41 @@ def validate_values(graph):
                             (instance, on_property, some_values_from)
                         )
     return missing_triples
+
+
+def parse_workflow(
+    workflow,
+    NS,
+    graph=None,
+    inherit_properties=True,
+    hasNode=None,
+    hasSourceFunction=None,
+    hasUnits=None,
+    inheritsPropertiesFrom=None,
+):
+    if hasNode is None:
+        hasNode = NS.hasNode
+    if hasSourceFunction is None:
+        hasSourceFunction = NS.hasSourceFunction
+    if hasUnits is None:
+        hasUnits = NS.hasUnits
+    if inheritsPropertiesFrom is None:
+        inheritsPropertiesFrom = NS.inheritsPropertiesFrom
+    if graph is None:
+        graph = Graph()
+    workflow_label = NS[workflow.label]
+    graph.add((workflow_label, RDFS.label, Literal(workflow.label)))
+    for value in workflow.children.values():
+        data = get_inputs_and_outputs(value)
+        graph.add((workflow_label, hasNode, NS[workflow.label + "." + data["label"]]))
+        graph += get_triples(
+            data=data,
+            NS=NS,
+            workflow_namespace=workflow.label,
```

Comment:
Maybe I'm misreading `rdflib.Namespace`, but couldn't we get rid of the `workflow_namespace` variable entirely and just use `NS=Namespace(NS + "." + workflow.label)`?

Reply:
Ah nevermind, I see in `get_triples` that screws up `label_def_triple = (NS[full_label], hasSourceFunction, NS[data["function"]])` where you want both the root scope and the local workflow scope.

This is my first exposure to `rdflib` and so far I'm not super happy with it, but I feel like despite its awkward feeling to me there must be a way to eek out slightly more elegance in how we handle the scoping here. Sorry I don't have something more constructive right now, just this gut feeling.